contextual_lenses/pfam_utils.py

```diff
@@ -31,201 +31,265 @@
 
 # Data preprocessing.
 # Original code source: https://www.kaggle.com/drewbryant/starter-pfam-seed-random-split.
-data_partitions_dirpath = 'random_split/'
-bucket_name = 'sequin-public'
+def read_all_shards(partition, data_dir, bucket_name):
+    """Combines different CSVs into a single dataframe."""
 
-def read_all_shards(partition='train', data_dir=data_partitions_dirpath, bucket_name=bucket_name):
-  """Combines different CSVs into a single dataframe."""
+    gcsfs = GCSFS(bucket_name)
 
-  gcsfs = GCSFS(bucket_name)
-
-  shards = []
-  for fn in gcsfs.listdir(data_partitions_dirpath + partition):
-    with gcsfs.open(os.path.join(data_partitions_dirpath, partition, fn)) as f:
-      shards.append(pd.read_csv(f, index_col=None))
-  return pd.concat(shards)
+    shards = []
+    for fn in gcsfs.listdir(data_dir + partition):
+        with gcsfs.open(os.path.join(data_dir, partition, fn)) as f:
+            shards.append(pd.read_csv(f, index_col=None))
+    return pd.concat(shards)
 
 
 def mod_family_accession(family_accession):
-  """Reduces family accession to everything prior to '.'."""
+    """Reduces family accession to everything prior to '.'."""
+
+    return family_accession[:family_accession.index('.')]
 
-  return family_accession[:family_accession.index('.')]
 
 # Padding.
 pfam_protein_domain = domains.VariableLengthDiscreteDomain(
-  vocab=domains.ProteinVocab(
-    include_anomalous_amino_acids=True,
-    include_bos=True,
-    include_eos=True,
-    include_pad=True,
-    include_mask=True),
-  length=512)
+    vocab=domains.ProteinVocab(include_anomalous_amino_acids=True,
+                               include_bos=True,
+                               include_eos=True,
+                               include_pad=True,
+                               include_mask=True),
+    length=512)
 
 pfam_num_categories = 27
 
+
 def residues_to_one_hot_inds(seq):
-  """Converts amino acid residues to one hot indices."""
+    """Converts amino acid residues to one hot indices."""
 
-  one_hot_inds = pfam_protein_domain.encode([seq])[0]
-    
-  return one_hot_inds
+    one_hot_inds = pfam_protein_domain.encode([seq])[0]
+
+    return one_hot_inds
 
 
 # Dictionary mapping family id to index.
-family_ids = open(resource_filename('contextual_lenses.resources', 'pfam_family_ids.txt'), 'r').readlines()
+family_ids = open(
```

Comment:
want to avoid doing compute/io heavy things on import - instead call as a function to load.

Suggest making a `load_family_ids` function which you call inside `create_pfam_df` to load.  Can also add a functools.lru_cache to avoid reloading.